### PR TITLE
Cleanup Testgrid a bit

### PR DESCRIPTION
**Pull request by adrcunha**
* rename "release" tab to "nightly" (this is a more appropriate and clear name)
* add back Testgrid config test that was dropped by mistake in #583
* remove the now unused api-coverage job (and tab)

### CODE REVIEW

File: ci/prow/make_config.go [hunk 1]
```diff
@@ -1349,48 +1350,37 @@ func executeTestGroupTemplate(testGroupName string, gcsLogDir string, extras map
...
+			case "performance":
 				baseOptions = testgridTabGroupByTarget
-			}
-			if jobName == "latency" {
+			case "latency":
+				baseOptions = testgridTabGroupByDir
 				extras["description"] = "95% latency in ms"
-			}
-			if jobName == "api-coverage" {
+			case "api-coverage":
```
Comment (srinivashegde86) [inline ci/prow/make_config.go]: we recently removed api-coverage as a tab as it was fialing and we did not want to invest looking into it

Comment (adrcunha) [reply ci/prow/make_config.go]: Unrelated, but done. :P

**Comment by adrcunha** [comment]
/hold cancel

We'll need another LGTM due to the changes.

**Pull request merged**